src/modules/cpu/host_advanced_augmentations.hpp

```diff
@@ -34,12 +34,13 @@ RppStatus water_host_batch(T* srcPtr, RppiSize *batch_srcSize, RppiSize *batch_s
                          Rpp32f *batch_freq_x, Rpp32f *batch_freq_y,
                          Rpp32f *batch_phase_x, Rpp32f *batch_phase_y,
                          Rpp32u outputFormatToggle, Rpp32u nbatchSize,
-                         RppiChnFormat chnFormat, Rpp32u channel)
+                         RppiChnFormat chnFormat, Rpp32u channel, rpp::Handle& handle)
 {
+    Rpp32u numThreads = handle.GetNumThreads();
```

Comment:
Done
We have made changes to limit the maximum value for numThreads to max number of hardware cpu threads available in the system
https://github.com/r-abishek/rpp/blob/ar/omp_threads_update/src/modules/hip/handlehip.cpp#L285

We already had a check where if  numThreads = 0 it will be set to batchSize
Valid range for numthreads is (0, max hardware threads in system]